neo/Core/TX/Transaction.py

```diff
@@ -716,7 +716,9 @@ def GetScriptHashesForVerifying(self):
                     hashes.add(UInt160(data=attr.Data))
 
         for key, group in groupby(self.outputs, lambda p: p.AssetId):
-            asset = GetBlockchain().GetAssetState(key.ToBytes())
+            asset = Helper.StaticAssetState(key)
```

Comment:
This will prevent any new assets registered to the blockchain to not function unless we update the `StaticAssetState()` helper function, which is not desirable.  What we can do is add an internal flag to the base TX class, which you set specifically and only when building raw transactions, and `GetScriptHashesForVerifying` can check the flag to determine if it will use the `StaticChecker` or the dynamic.

Reply:
Okay, that makes sense. However, wouldn't the StaticChecker also need to be updated if new assets were added? I am assuming the static checker would verify the input hash vs NEO/GAS.

Reply:
Yes that's a good point. I would rather see it failing though in an isolated case (raw transactions) than in a generic case. I don't know the full extend of the code base of what else will break by doing this. To me and with the knowledge I currently have this sounds like the safest approach while maintaining backwards compatibility.

Reply:
I am still not sure I understand, but I am implementing a concept with a new flag. In my original method, if the input hash doesn't match NEO or GAS for the comparison, it will return `None` at which point the normal process will proceed.